api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/ContrastAccountServiceImpl.java

```diff
@@ -0,0 +1,238 @@
+        AccountValidationResponse validationResponse = new AccountValidationResponse();
+        validationResponse.setValidationStatus(FAILURE);
+        validationResponse.setMessage(VALIDATION_FAILED);
+        try {
+            String apiUrl = new URIBuilder(String.format(CONTRAST_URL_TEMPLATE, pluginData.getContrastEnvironmentName()))
+                    .setPath(ORGANIZATIONS_URL_PATH + pluginData.getContrastOrganizationId()).build().toString();
+            String authKey = Base64.getEncoder().encodeToString((pluginData.getContrastUserId() + ":"
+                    + pluginData.getContrastServiceKey()).getBytes());
+            HttpGet request = new HttpGet(apiUrl);
+            request.setHeader(HttpHeaders.AUTHORIZATION, authKey);
+            request.setHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE);
+            request.setHeader("API-Key", pluginData.getContrastApiKey());
+            try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+                try (CloseableHttpResponse response = httpClient.execute(request)) {
+                    int statusCode = response.getStatusLine().getStatusCode();
+                    if (statusCode != 200) {
+                        String errorMessage = handleErrorResponse(response.getStatusLine().getStatusCode());
+                        validationResponse.setErrorDetails(errorMessage);
+                        return validationResponse;
+                    }
+                    JsonNode jsonResponse = objectMapper.readTree(response.getEntity().getContent());
+                    String organizationName = jsonResponse.get("name").textValue();
+                    if (StringUtils.isEmpty(organizationName)) {
+                        validationResponse.setErrorDetails("Couldn't read Organization name");
+                        return validationResponse;
+                    }
+                    pluginData.setContrastOrganizationName(organizationName);
+                    validationResponse.setValidationStatus(SUCCESS);
+                    validationResponse.setErrorDetails(null);
+                    validationResponse.setMessage("Connection established successfully");
+                    return validationResponse;
+                }
+            }
+        } catch (UnknownHostException e) {
+            LOGGER.error("Unknown host error: " + e.getMessage(), e);
+            validationResponse.setErrorDetails("The Environment name may be incorrect or the server may be down.");
+        } catch (Exception e) {
+            LOGGER.error("Unexpected error: " + e.getMessage(), e);
+            validationResponse.setErrorDetails("Unexpected error occurred");
+        }
+        return validationResponse;
+    }
+
+    private String handleErrorResponse(int statusCode) {
+        switch (statusCode) {
+            case 400:
+            case 404:
+                return "Bad request: Check the Organization ID and Keys and make sure they are correct.";
```

Comment:
From my experience 404 has nothing to do with auth hence with Keys

Reply:
yeah, 404 occurs when Organization ID is wrong and when the contrast server is down